### PR TITLE
List the things that happen when you deposit a work

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: pages/hydra/developer_resources/what-happens-deposit.md [hunk 1]
```diff
@@ -7,3 +7,15 @@ folder: hydra/how-to/
 sidebar: home_sidebar
 tags: [development_resources]
 ---
+
+
+1. I'm on the form to create a new work and I click the submit.
+1. The data is submitted to the controller for the correct work type.  This controller was generated when you did `bin/rails generate hyrax:work <type>`.  For example the `app/controllers/hyrax/generic_works_controller.rb` if you generated a work type of `GenericWork`.
+1. This controller is including `Hyrax::WorksControllerBehavior`, so the `create` method is run: https://github.com/projecthydra-labs/hyrax/blob/master/app/controllers/concerns/hyrax/works_controller_behavior.rb#L57-L69
```
Comment (elrayle) [inline pages/hydra/developer_resources/what-happens-deposit.md]: Can you make clickable links?  You would think the second one would be automatically, but it isn't.  
```
[Hyrax::WorksControllerBehavior](https://github.com/projecthydra-labs/hyrax/blob/master/app/controllers/concerns/hyrax/works_controller_behavior.rb)
[create](https://github.com/projecthydra-labs/hyrax/blob/master/app/controllers/concerns/hyrax/works_controller_behavior.rb#L57-L69)
```

File: pages/hydra/developer_resources/what-happens-deposit.md [hunk 2]
```diff
@@ -7,3 +7,15 @@ folder: hydra/how-to/
...
+1. The `create` method initializes the actor https://github.com/projecthydra-labs/hyrax/blob/master/app/controllers/concerns/hyrax/works_controller_behavior.rb#L139 and calls create on it.
```
Comment (elrayle) [inline pages/hydra/developer_resources/what-happens-deposit.md]:
```
[initializes the actor](https://github.com/projecthydra-labs/hyrax/blob/master/app/controllers/concerns/hyrax/works_controller_behavior.rb#L139)
```

File: pages/hydra/developer_resources/what-happens-deposit.md [hunk 3]
```diff
@@ -7,3 +7,15 @@ folder: hydra/how-to/
...
+1. The actor is a stack built of functions that take parts of the form data and act upon them.  The actor is customizable in your local application, but by default the steps of the pipeline are defined here: https://github.com/projecthydra-labs/hyrax/blob/master/app/services/hyrax/default_middleware_stack.rb#L4-L21
```
Comment (elrayle) [inline pages/hydra/developer_resources/what-happens-deposit.md]: with the default steps of the pipeline defined by `[default_middleware_stack](https://github.com/projecthydra-labs/hyrax/blob/master/app/services/hyrax/default_middleware_stack.rb#L4-L21)`.

File: pages/hydra/developer_resources/what-happens-deposit.md [hunk 4]
```diff
@@ -7,3 +7,15 @@ folder: hydra/how-to/
...
+1. When create is called, each frame of this stack is called, and passed a Hyrax::Environment object which contains the new work, the current user and the data from the HTML form.
```
Comment (elrayle) [inline pages/hydra/developer_resources/what-happens-deposit.md]: link to Hyrax::Environment

File: pages/hydra/developer_resources/what-happens-deposit.md [hunk 5]
```diff
@@ -7,3 +7,15 @@ folder: hydra/how-to/
...
+1. The `create` method initializes the actor https://github.com/projecthydra-labs/hyrax/blob/master/app/controllers/concerns/hyrax/works_controller_behavior.rb#L139 and calls create on it.
+1. The actor is a stack built of functions that take parts of the form data and act upon them.  The actor is customizable in your local application, but by default the steps of the pipeline are defined here: https://github.com/projecthydra-labs/hyrax/blob/master/app/services/hyrax/default_middleware_stack.rb#L4-L21
+1. When create is called, each frame of this stack is called, and passed a Hyrax::Environment object which contains the new work, the current user and the data from the HTML form.  
+1. one of the frames in the stack is the https://github.com/projecthydra-labs/hyrax/blob/master/app/actors/hyrax/actors/model_actor.rb. It calls the generated actor specific for the work type we're making. In this case `app/actors/hyrax/actors/generic_work_actor.rb`.  This generated actor extends `Hyrax::Actors::BaseActor`, which is responsible for persisting the object by calling the `ActiveFedora::Base#save` method.
```
Comment (elrayle) [inline pages/hydra/developer_resources/what-happens-deposit.md]: One of the frames in the stack is the [actors/hyrax/actors/model_actor.rb](https://github.com/projecthydra-labs/hyrax/blob/master/app/actors/hyrax/actors/model_actor.rb).

Link to Hyrax::Actors::BaseActor and ActiveFedora::Base#save

File: pages/hydra/developer_resources/what-happens-deposit.md [hunk 6]
```diff
@@ -7,3 +7,15 @@ folder: hydra/how-to/
...
+1. Another frame in the stack was the `Hyrax::Actors::InitializeWorkflowActor`.  This actor creates a workflow entity proxy for the object. This proxy tracks where the work is in the workflow.
```
Comment (elrayle) [inline pages/hydra/developer_resources/what-happens-deposit.md]: Link to Hyrax::Actors::InitializeWorkflowActor

**Comment by elrayle** [comment]
I tried to add links to files in 1-0-stable branch.  The actor does not have a create method in 1.0.  This was added in master for 2.0.  Since 2.0 isn't complete yet, I prefer the instructions be accurate for 1.0.  Will require more investigation to modify instructions for 1.0 code.

**Comment by elrayle** [comment]
There are two pages for **What happens when... I deposit something**.  One is for 1-0-stable and the other is for master (2.0.alpha).  I also moved all the What happens when docs to a set of subfolders under heading **What happens when...**

I would like to continue to work on how we have documentation for multiple versions.  Right now, there are two listings in the sidebar menu, one for each version.  Long term, that will become unwieldy.  One possible solution is to have one link in the sidebar and a standard header on the page that links to alternate versions.

**Comment by elrayle** [comment]
@jcoyne Want to take a look at the 1.0 instructions to see if you agree?

**Comment by jcoyne** [comment]
Looks good except the URLs need to change from `samvera-labs/` to `samvera/` See: https://github.com/samvera/samvera.github.io/pull/90

**Comment by elrayle** [comment]
Per issue #90, links were updated from samvera-labs to samvera.

**Pull request merged**